lib/IRGen/GenCoverage.cpp

```diff
@@ -112,13 +128,21 @@ void IRGenModule::emitCoverageMaps(ArrayRef<const SILCoverageMap *> Mappings) {
     const uint64_t NameHash = llvm::IndexedInstrProf::ComputeHash(NameValue);
     std::string FuncRecordName = "__covrec_" + llvm::utohexstr(NameHash);
 
-    unsigned FileID =
-        std::find(Files.begin(), Files.end(), M->getFilename()) - Files.begin();
+    // The file ID needs to be bumped by 1 to account for the working directory
+    // as the first element.
+    unsigned FileID = 1 +
+                      std::find(Files.begin(), Files.end(), M->getFilename()) -
+                      Files.begin();
+    assert(FileID < FilenameStrs.size());
```

Comment:
Now that I think about it, it really doesn't seem great that we're using `std::find` here, it'll be fine for cases where we have N output files for N input files (since we shard the coverage maps, so `Files` will be max 1 element), but for WMO, this will be O(N^2) for the number of files....